pkg/assets/create/creater.go

```diff
@@ -163,36 +163,58 @@ func create(ctx context.Context, manifests map[string]*unstructured.Unstructured
 			continue
 		}
 
+		var resource dynamic.ResourceInterface
 		if mappings.Scope.Name() == meta.RESTScopeNameRoot {
-			_, err = client.Resource(mappings.Resource).Create(manifests[path], metav1.CreateOptions{})
+			resource = client.Resource(mappings.Resource)
 		} else {
-			_, err = client.Resource(mappings.Resource).Namespace(manifests[path].GetNamespace()).Create(manifests[path], metav1.CreateOptions{})
+			resource = client.Resource(mappings.Resource).Namespace(manifests[path].GetNamespace())
 		}
-
 		resourceString := mappings.Resource.Resource + "." + mappings.Resource.Version + "." + mappings.Resource.Group + "/" + manifests[path].GetName() + " -n " + manifests[path].GetNamespace()
 
+		incluster, err := resource.Create(manifests[path], metav1.CreateOptions{})
+
+		if err == nil && options.Verbose {
+			fmt.Fprintf(options.StdErr, "Created %q %s\n", path, resourceString)
+		}
+
 		// Resource already exists means we already succeeded
 		// This should never happen as we remove already created items from the manifest list, unless the resource existed beforehand.
 		if kerrors.IsAlreadyExists(err) {
 			if options.Verbose {
 				fmt.Fprintf(options.StdErr, "Skipped %q %s as it already exists\n", path, resourceString)
 			}
-			delete(manifests, path)
-			continue
+			incluster, err = resource.Get(manifests[path].GetName(), metav1.GetOptions{})
+			if err != nil {
+				if options.Verbose {
+					fmt.Fprintf(options.StdErr, "Failed to get already existing %q %s: %v\n", path, resourceString, err)
+				}
+				errs[path] = fmt.Errorf("failed to get %s: %v", resourceString, err)
+				continue
+			}
 		}
 
 		if err != nil {
 			if options.Verbose {
 				fmt.Fprintf(options.StdErr, "Failed to create %q %s: %v\n", path, resourceString, err)
 			}
-			errs[path] = fmt.Errorf("failed to create: %v", err)
+			errs[path] = fmt.Errorf("failed to create %s: %v", resourceString, err)
 			continue
 		}
 
-		if options.Verbose {
-			fmt.Fprintf(options.StdErr, "Created %q %s\n", path, resourceString)
+		if _, ok := manifests[path].Object["status"]; ok {
```

Comment:
I would expect a check that `include.Object` has no status field. Otherwise, we might race with other actors setting status.

Reply:
Hmm, i'm not sure I follow your suggestion? how does that prevent race... ?

Reply:
an manifests are file on disk, why would they change mid-execution?

Reply:
You creation of object has two step: main create call plus status update. If there is time in between some operator waiting for the object might be update it in between.

Maybe a more theoretical problem. But just checking for /status wouldn't harm I think.

Reply:
> If there is time in between some operator waiting for the object might be update it in between.
> Maybe a more theoretical problem. But just checking for .status wouldn't harm I think.

So if an operator in the mean time updated the status, cluster-bootstrap should skip UpdateStatus is that what you are suggesting?

Reply:
Yes.

Note this code is potentially not only used in cluster-bootstrap. It is better generically correct.

Reply:
@sttts updated to skip UpdateStatus when it already exists incluster.. with https://github.com/openshift/library-go/pull/479/commits/30c07c12fcd1bc3ee2093c1558985d154b89c995